### PR TITLE
Add shell command for clearing caches (refs #7368)

### PR DESCRIPTION
**Pull request by chris48s**
Add shell command for clearing caches for use in deployment scripts, etc.
See discussion in issue #7368

### CODE REVIEW

File: src/Shell/CacheShell.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+     * @param string|null $prefix The cache prefix to be cleared.
+     * @throws Cake\Console\Exception\StopException
+     * @return void
+     */
+    public function clear($prefix = null)
+    {
+        try {
+            $engine = Cache::engine($prefix);
+            Cache::clear(false, $prefix);
+            if ($engine instanceof \Cake\Cache\Engine\ApcEngine) {
```
Comment (ADmad) [inline src/Shell/CacheShell.php]: Instead of using FQCN add `use` statements for the classes at the top.

Comment (markstory) [inline src/Shell/CacheShell.php]: :+1:

File: src/Shell/CacheShell.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        $parser->addSubcommand('clear_all', [
+            'help' => 'Clear all caches.',
+        ]);
+        $parser->addSubcommand('clear', [
+            'help' => 'Clear the cache for a specified prefix.',
+            'parser' => [
+                'description' => [
+                    'Clear the cache for a particular prefix.',
+                    'For example, `cake cache clear _cake_model_` will clear the model cache',
+                    'Use `cake cache clear list_prefixes` to list available prefixes'
```
Comment (markstory) [inline src/Shell/CacheShell.php]: Does `cake cache clear list_prefixes` actually work?

**Comment by chris48s** [comment]
Updated based on comments

**Comment by lorenzo** [comment]
This looks good to me 👍

**Pull request merged**